server/src/queues/createApp.ts

```diff
@@ -0,0 +1,138 @@
+import { Worker, Queue } from 'bullmq';
+import createDebug from 'debug';
+import Redis from 'ioredis';
+import { pubsub } from './../index';
+import { config } from '../config';
+import { sshConnect } from '../lib/ssh';
+import { dokku } from '../lib/dokku';
+import { prisma } from '../prisma';
+import fetch from 'node-fetch';
+
+const queueName = 'create-app';
+const debug = createDebug(`queue:${queueName}`);
+const redisClient = new Redis(config.redisUrl);
+
+interface QueueArgs {
+  appName: string;
+  gitRepoUrl: string;
+  userId: string;
+  branchName?: string;
+}
+
+export const createAppQueue = new Queue<QueueArgs>(queueName, {
+  defaultJobOptions: {
+    // Max timeout 20 minutes
+    timeout: 1.2e6,
+  },
+  connection: redisClient,
+});
+
+/**
+ * - Create app
+ */
+const worker = new Worker(
+  queueName,
+  async (job) => {
+    const { appName, gitRepoUrl, userId, branchName } = job.data;
+
+    debug(
+      `starting create app queue for ${userId} and ${appName} app with repoURL ${gitRepoUrl} and branch name ${branchName}`
+    );
+
+    const branch = branchName ? branchName : 'main';
+
+    const ssh = await sshConnect();
+
+    //to do move this out to UTILS
+    const getRepoData = (gitRepoUrl: string) => {
```

Comment:
lets move this logic in the mutation, the validation should be done before the queue start in case the user send a non existing repo for example